### PR TITLE
[FEDE-6613] Upgrade arrow to 15.0.0

### PR DESCRIPTION
**Pull request by prateeknima77**

### CODE REVIEW

File: java/adapter/orc/pom.xml [hunk 1]
```diff
@@ -111,7 +111,7 @@
     <parent>
         <groupId>org.apache.arrow</groupId>
         <artifactId>arrow-java-root</artifactId>
-        <version>15.0.0</version>
+        <version>siren-15.0.0-4-SNAPSHOT</version>
```
Comment (GeorgeAp) [inline java/adapter/orc/pom.xml]: Should this be version `siren-15.0.0-4-SNAPSHOT` ?

File: java/bom/pom.xml [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
   <groupId>org.apache.arrow</groupId>
   <artifactId>arrow-bom</artifactId>
-  <version>15.0.0</version>
+  <version>siren-15.0.0</version>
```
Comment (GeorgeAp) [inline java/bom/pom.xml]: why this change, can you please apply the patch?

File: java/c/pom.xml [hunk 3]
```diff
@@ -13,7 +13,7 @@
     <parent>
         <artifactId>arrow-java-root</artifactId>
         <groupId>org.apache.arrow</groupId>
-        <version>15.0.0</version>
+        <version>siren-15.0.0-4-SNAPSHOT</version>
```
Comment (GeorgeAp) [inline java/c/pom.xml]: do we use this?

**Review by GeorgeAp — CHANGES_REQUESTED** [review]
Can you please check the versions and compare the cherry pick

File: java/dataset/pom.xml [hunk 4]
```diff
@@ -15,7 +15,7 @@
     <parent>
         <artifactId>arrow-java-root</artifactId>
         <groupId>org.apache.arrow</groupId>
-        <version>15.0.0</version>
+        <version>siren-15.0.0-4-SNAPSHOT</version>
```
Comment (GeorgeAp) [inline java/dataset/pom.xml]: do we use this?

File: java/vector/src/main/java/org/apache/arrow/vector/BitVectorHelper.java [hunk 5]
```diff
@@ -18,6 +18,9 @@
 package org.apache.arrow.vector;
 
 import static org.apache.arrow.memory.util.LargeMemoryUtil.checkedCastToInt;
+import static siren.io.netty.util.internal.PlatformDependent.getByte;
+import static siren.io.netty.util.internal.PlatformDependent.getInt;
+import static siren.io.netty.util.internal.PlatformDependent.getLong;
```
Comment (GeorgeAp) [inline java/vector/src/main/java/org/apache/arrow/vector/BitVectorHelper.java]: These can be removed, `PlatformDependent` is not used in arrow vector anymore, see https://github.com/apache/arrow/pull/14913

File: README.md [hunk 6]
```diff
@@ -17,6 +17,74 @@
...
+```sh
+$ cd java
+$ mvn clean package
+```
+
+Because of the default value change of `unsafe_memory_access` property, some
+tests in `vector` fail.
+
+```sh
+mvn -pl memory,memory/memory-core,memory/memory-netty,memory/memory-unsafe,format,vector install -Dsiren.arrow.enable_unsafe_memory_access=false -Dsiren.drill.enable_unsafe_memory_access=false
```
Comment (GeorgeAp) [inline README.md]: can you update this with the new command ? `mvn clean -pl maven,maven/module-info-compiler-maven-plugin,memory,memory/memory-core,memory/memory-netty,memory/memory-unsafe,format,vector package -Dsiren.arrow.enable_unsafe_memory_access=false -Dsiren.drill.enable_unsafe_memory_access=false`

**Review by GeorgeAp — CHANGES_REQUESTED** [review]

File: java/pom.xml [hunk 7]
```diff
@@ -33,7 +33,7 @@
     <dep.junit.jupiter.version>5.10.1</dep.junit.jupiter.version>
     <dep.slf4j.version>2.0.9</dep.slf4j.version>
     <dep.guava-bom.version>33.0.0-jre</dep.guava-bom.version>
-    <dep.netty-bom.version>4.1.104.Final</dep.netty-bom.version>
+    <dep.netty-bom.version>siren-4.1.104-2</dep.netty-bom.version>
```
Comment (GeorgeAp) [inline java/pom.xml]:
```suggestion
    <dep.netty-bom.version>siren-4.1.104-1-SNAPSHOT</dep.netty-bom.version>
```

File: README.md (hunk 6 above)
Comment (prateeknima77) [reply README.md]: Done: adfeb461d70ecd6f5dc5635cecf667c1ea7a987b

File: java/pom.xml (hunk 7 above)
Comment (prateeknima77) [reply java/pom.xml]: The netty version has been updated: c9f0cc0fe0fd94ce19c45c94fcfcb0433fe9a936

File: java/vector/src/main/java/org/apache/arrow/vector/BitVectorHelper.java (hunk 5 above)
Comment (prateeknima77) [reply java/vector/src/main/java/org/apache/arrow/vector/BitVectorHelper.java]: Yes, thats correct. This has been fixed: 77b48e98d6c1c6da85d9cc9bcbd11a1f2f2eda63

File: java/bom/pom.xml (hunk 2 above)
Comment (prateeknima77) [reply java/bom/pom.xml]: The version has been fixed: 51d8de67c9bce8ac4e1886244d6f94246d888924

File: java/adapter/orc/pom.xml (hunk 1 above)
Comment (prateeknima77) [reply java/adapter/orc/pom.xml]: Fixed snapshot versions: 33c96572bae9bec60b0109500f0fab0a1cb827f9

File: java/c/pom.xml (hunk 3 above)
Comment (prateeknima77) [reply java/c/pom.xml]: Fixed snapshot versions: 33c96572bae9bec60b0109500f0fab0a1cb827f9

File: java/dataset/pom.xml (hunk 4 above)
Comment (prateeknima77) [reply java/dataset/pom.xml]: Fixed snapshot versions: 33c96572bae9bec60b0109500f0fab0a1cb827f9

File: java/c/pom.xml (hunk 3 above)
Comment (GeorgeAp) [reply java/c/pom.xml]: We don't use this, it can be reverted

File: java/dataset/pom.xml (hunk 4 above)
Comment (GeorgeAp) [reply java/dataset/pom.xml]: We don't use this, it can be reverted

File: cpp/submodules/parquet-testing
Comment (GeorgeAp) [inline cpp/submodules/parquet-testing:1]: Can this be removed?

File: testing
Comment (GeorgeAp) [inline testing:1]: Can this be removed?

**Review by GeorgeAp — CHANGES_REQUESTED** [review]

File: java/c/pom.xml (hunk 3 above)
Comment (GeorgeAp) [reply java/c/pom.xml]: @scampi this arrow version in this pom was updated to  `siren-11.0.0-3-SNAPSHOT` in the pr https://github.com/sirensolutions/arrow/pull/48 but we do not use it. Is there a plan to use it or we can revert the change ?

File: java/dataset/pom.xml (hunk 4 above)
Comment (GeorgeAp) [reply java/dataset/pom.xml]: @scampi it is the same comment here

File: java/c/pom.xml (hunk 3 above)
Comment (scampi) [reply java/c/pom.xml]: I think it can be reverted. This project provides C interfaces but we don't use that.

File: java/dataset/pom.xml (hunk 4 above)
Comment (scampi) [reply java/dataset/pom.xml]: it can be reverted

File: java/c/pom.xml (hunk 3 above)
Comment (prateeknima77) [reply java/c/pom.xml]: The changes have been reverted: 3c8bcca1146217fe75bafd17e513787be9c62e4f

File: java/dataset/pom.xml (hunk 4 above)
Comment (prateeknima77) [reply java/dataset/pom.xml]: The changes have been reverted: 3c8bcca1146217fe75bafd17e513787be9c62e4f
